src/main/scala/org/broadinstitute/dsde/workbench/avram/db/CollectionComponent.scala

```diff
@@ -35,10 +36,15 @@ trait CollectionComponent extends AvramComponent {
 
   object collectionQuery extends TableQuery(new CollectionTable(_)) {
 
-    def save(externalId: UUID, samResource: String, createdBy: String): DBIO[Int] = {
+    def save(externalId: UUID, samResource: SamResource, createdBy: String)(implicit executionContext: ExecutionContext): DBIO[Collection] = {
       val now = Timestamp.from(Instant.now)
       //currently the updatedBy and updatedTimestamp are the same as created at save time
-      collectionQuery += CollectionRecord(0, externalId, samResource, createdBy, now, createdBy, now)
+      val collectionRecord = CollectionRecord(0, externalId, samResource.resourceName, createdBy, now, createdBy, now)
+      for {
+        _ <- collectionQuery += collectionRecord
+      } yield {
+        unmarshalCollection(collectionRecord)
```

Comment:
I guess this would unmarshal a `CollectionRecord` where the id is 0. Is there anyway to insert to the DB and return the id that was autogenerated? I think postgres has an `INSERT ... RETURNING ...`, else we could do a insert + select.

Reply:
Maybe something like this: https://stackoverflow.com/a/31448129/244191